mobile/library/swift/stats/Element.swift

```diff
@@ -9,13 +9,27 @@ public final class Element: NSObject, ExpressibleByStringLiteral {
   internal let value: String
 
   public init(stringLiteral value: String) {
-    guard value.range(of: kPattern, options: .regularExpression) != nil else {
-      preconditionFailure("Element values must conform to the regex '\(kPattern)'.")
-    }
+    precondition(
+      value.matchesStatsElementPattern,
+      "Element values must conform to the regex '\(kPattern)'."
+    )
     self.value = value
   }
 
   public override func isEqual(_ object: Any?) -> Bool {
     return (object as? Element)?.value == self.value
   }
 }
+
+extension String {
+  var matchesStatsElementPattern: Bool {
+    if #available(iOS 16.0, macOS 13.0, *) {
+      return self.contains(/^[A-Za-z_]+$/)
+    }
+#if canImport(std)
```

Comment:
can we add a comment saying when it's expected to be true versus false?

Reply:
Sure, I'll add a comment. `std` is the name of the C++ stdlib when importing it into Swift. So this checks if we're compiling with `-enable-experimental-cxx-interop`.